internal/config/storage.go

```diff
@@ -0,0 +1,43 @@
+package config
+
+import (
+	"github.com/spf13/viper"
+)
+
+type StorageType string
+
+const (
+	DatabaseStorageType = StorageType("database")
+	LocalStorageType    = StorageType("local")
+	GitStorageType      = StorageType("git")
+)
+
+// StorageConfig contains fields which will configure the type of backend in which Flipt will serve
+// flag state.
+type StorageConfig struct {
+	Type  StorageType `json:"type,omitempty" mapstructure:"type"`
+	Local Local       `json:"local,omitempty" mapstructure:"local"`
+	Git   Git         `json:"git,omitempty" mapstructure:"git"`
+}
+
+func (c *StorageConfig) setDefaults(v *viper.Viper) {
+	v.SetDefault("storage", map[string]any{
+		"type": "database",
+		"local": map[string]any{
+			"path": "",
+		},
+		"git": map[string]any{
+			"repository": "",
+			"ref":        "",
+		},
+	})
+}
```

Comment:
Reflecting on the POC implementation there is a bug in it. But I think it would be nice to keep these defaults to be conditional. Otherwise, we default a bunch of things in all cases.
See the original example here: https://github.com/flipt-io/flipt/blob/gm/filesystem-backends/internal/config/storage.go#L30-L36

I think we could do this correctly like so:

```suggestion
func (c *StorageConfig) setDefaults(v *viper.Viper) {
	v.SetDefault("type", "database")

	switch v.GetString("type") {
	case string(LocalStorageType):
		v.SetDefault("local", map[string]any{
			"path": ".",
		})
	case string(GitStorageType):
		v.SetDefault("git", map[string]any{
			"ref":  "main",
		})
	}
}
```

Reply:
@GeorgeMac Addressed! Thanks it looks like to reference nested structures you have to use the "." as a delimiter.